raft/src/main/java/org/apache/kafka/raft/RaftConfig.java

```diff
@@ -76,7 +87,48 @@
     private final int electionBackoffMaxMs;
     private final int fetchTimeoutMs;
     private final int appendLingerMs;
-    private final Map<Integer, InetSocketAddress> voterConnections;
+    private final Map<Integer, AddressSpec> voterConnections;
+
+    public static abstract class AddressSpec {
+       public abstract InetSocketAddress address();
```

Comment:
Do we need this in the abstract class? I was thinking we would only be able to access `InetSocketAddress` if the type is `InetAddressSpec`. Otherwise the type protection from `AddressSpec` loses its bite.

Reply:
Yea, I did a second pass later and figured the address defeats the purpose of the type enforcement (especially returning an InetSocketAddress for any address spec). I reverted to an interface and have only InetSocketAddressSpec contain an InetSocketAddress